### DISCUSSION TITLE
How sync with on .git folder (root project) and change it on switch between files?

### DISCUSSION THREAD

**Discussion by nitaicharan** [nvim-tree/nvim-tree.lua#2036] [Q&A] [answered]
I'm trying to set up nvim-tree to change the root folder with the project root folder (with .git). Which `setup` properties do I have to turn on?
I've tried many ways, but I couldn't achieve this behavior.Follow my config:
```lua
  sync_root_with_cwd = true,
  reload_on_bufenter = true,
  prefer_startup_root = false,
  view = {
    mappings = {
      custom_only = true,
      list = {
        { key = "d",           action = "remove" },
        { key = "h",           action = "parent_node" },
        { key = "H",           action = "close_node" },
        { key = "R",           action = "rename" },
        { key = "m",           action = "bulk_move" },
        { key = "p",           action = "paste" },
        { key = "P",           action = "preview" },
        { key = "cf",          action = "create" },
        { key = "gr",          action = "refresh" },
        { key = "oh",          action = "split" },
        { key = "ov",          action = "vsplit" },
        { key = "ox",          action = "system_open" },
        { key = "yf",          action = "copy" },
        { key = "ya",          action = "copy_absolute_path" },
        { key = "yp",          action = "copy_name" },
        { key = "yr",          action = "copy_path" },
        { key = { "oo", "l" }, action = "edit" },
        { key = "?",           action = "toggle_help" },
      },
    },
  },
  actions = {
    change_dir = {
      enable = true,
    },
  },
  renderer = {
    highlight_git = true,
    root_folder_label = ":t",
    icons = {
      show = {
        git = false,
      },
    },
  },
  update_focused_file = {
    enable = true,
    debounce_delay = 15,
    update_root = true,
  },
  diagnostics = {
    enable = true,
    show_on_open_dirs = true,
  },
  git = {
    ignore = false,
    timeout = 200,
  },
  trash = {
    cmd = "trash",
  },
```

**Comment by nitaicharan** [comment]
@alex-courtis I even tried your config, but I didn't get this event I want. Btw, thank you for all you've been doing for this project

**Reply by alex-courtis** [reply]
nvim-tree doesn't really have the concept of a project; it's just a file explorer.

You can define your "project roots" via `:help nvim-tree.root_dirs`. When you find files or use `:help nvim-tree.update_focused_file` it will use those. `:help nvim-tree.sync_root_with_cwd` is also useful as it will update the tree root when you `:cd ...`.

My workflow around tree root is that it (and vim's cwd) update when I explicitly change directories via `<C-]>` or `:cd`. When I want to change directory outside of the current I'll use `:NvimTreeFindFile!`

**Reply by nitaicharan** [reply]
Thank you for reply it.

**Comment by nitaicharan** [comment, marked as answer]
To make the working directory be synchronized with the parent folder with `.git` folder, you can use [ahmedkhalf/project.nvim](https://github.com/ahmedkhalf/project.nvim).

P.S.: I had this problem that my working directory was changing all the time I opened a new file. After a log time trying to figure out what was going on, my `null-ls.nvim` bellow had the active formatter with `prettier` and it was changing automatically the working directory for some reason:
```lua
local null_ls = require("null-ls")

null_ls.setup({
  debug = true,
  sources = {
    ...
    -- null_ls.builtins.formatting.prettier,
    ...
  },
})
```

This case can also happen, I guess, if the `autochdir` is active via command line `set autochdir` or via command file setting with `vim.o.autochdir = true`

**Reply by alex-courtis** [reply]
That plugin is rather nice.

To confirm: is nvim-tree is changing its root following project.nvim directory changes?

**Reply by gegoune** [reply]
Yes, it will. I have used that plugin in the past until I believe echanovsky posted few lines replacement on Reddit and since I was disabling lsp based functionally of project.nvim that was more than enough for me.

**Reply by nitaicharan** [reply]
Sorry I just saw it now. Yes @alex-courtis it's automatically change to the project root folder. Combine with nvim-tree feature of `update_focused_file` will show in the project root (which has .git folder).
